src/components/splat.js

```diff
@@ -0,0 +1,910 @@
+      bytesFileTotal = bytesFileTotal ? parseInt(bytesFileTotal) : undefined;
+
+      if (bytesFileTotal) {
+        var numVertices = Math.floor(bytesFileTotal / rowLength);
+        await this.initGL(numVertices);
+      }
+
+      var chunks = [];
+      var start = Date.now();
+      var lastReportedProgress = 0;
+
+      while (true) {
+        try {
+          var dataReceived = await reader.read();
+          if (dataReceived.done) {
+            console.log('Completed download.');
+            break;
+          }
+          var newChunk = dataReceived.value;
+          bytesDownloaded += newChunk.length;
+          chunks.push(newChunk);
+
+          // Downloar progress stats.
+          if (bytesFileTotal) {
+            var mbps = (bytesDownloaded / 1024 / 1024) / ((Date.now() - start) / 1000);
+            var percent = bytesDownloaded / bytesFileTotal * 100;
+            if (percent - lastReportedProgress > 1) {
+              console.log('Download progress:', percent.toFixed(2) + '%', mbps.toFixed(2) + ' Mbps');
+              lastReportedProgress = percent;
+            }
+          } else {
+            console.log('Download progress:', bytesDownloaded, ', unknown total');
+          }
+
+          var bytesToProcess = bytesDownloaded - bytesProcessed;
+          if (bytesFileTotal && bytesToProcess > rowLength) {
+            var concatenatedChunksbuffer = new Uint8Array(bytesToProcess);
+            var offset = 0;
+            for (var chunk of chunks) {
+              concatenatedChunksbuffer.set(chunk, offset);
+              offset += chunk.length;
+            }
+
+            var chunkBytesProcessed = this.processSplatChunk(concatenatedChunksbuffer, bytesToProcess);
+            bytesProcessed += chunkBytesProcessed;
+
+            // Reset chunks array. Copy leftover bytes if chunks not perfect multiple of a splat data structure.
+            chunks.length = 0;
+            if (bytesToProcess > chunkBytesProcessed) {
+              var extraData = new Uint8Array(bytesToProcess - chunkBytesProcessed);
+              extraData.set(concatenatedChunksbuffer.subarray(bytesToProcess - extraData.length, bytesToProcess), 0);
+              chunks.push(extraData);
+            }
+          }
+        } catch (error) {
+          console.error(error);
+          break;
+        }
+      }
+    },
+
+    processSplatChunk: function (chunksBuffer, chunkSize) {
+      var rowLength = 3 * 4 + 3 * 4 + 4 + 4;
+      var vertexCount = Math.floor(chunkSize / rowLength);
+      var bytesProcessed = vertexCount * rowLength;
+      var chunkBuffer = new Uint8Array(bytesProcessed);
+      chunkBuffer.set(chunksBuffer.subarray(0, chunkBuffer.byteLength), 0);
+      this.pushDataBuffer(chunkBuffer.buffer, vertexCount);
+      return bytesProcessed;
+    },
+
+    initGL: async function initGL (numVertices) {
+      console.log('initGL', numVertices);
+      this.el.object3D.frustumCulled = false;
```

Comment:
This doesn't seem necessary. The actual mesh has frustum culling disabled as well further down in this method.